src/foam/nanos/crunch/CrunchService.js

```diff
@@ -0,0 +1,49 @@
+foam.INTERFACE({
+  package: 'foam.nanos.crunch',
+  name: 'CrunchService',
+  documentation: `
+    CrunchService provides common logic used by the client and other CRUNCH
+    services.
+  `,
+
+  methods: [
+    {
+      name: 'getGrantPath',
+      documentation: `
+        getGrantPath provides a an array of capability objects representing
+        the list of capabilities required to grant the desired capability.
+      `,
+      async: true,
+      type: 'List',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'sourceId',
+          type: 'String'
+        }
+      ]
+    },
+    {
+      name: 'getJunction',
+      documentation: `
+        getJunction provides the correct UserCapabilityJunction based on the
+        context provided.
+      `,
+      async: true,
+      type: 'UserCapabilityJunction',
```

Comment:
Oh shoot, okay thanks for letting me know

Reply:
If you are pulling the context for ucj's that the business has with a bank account - then it would be a list.

Good catch @zahnno

Reply:
@Nauna We'll need to discuss this. Right now I don't know how the wizard is supposed to handle capabilities like this (checking the UCJ status and rendering the UCJ data), or how other code checking UCJ status is meant to do so.

Reply:
- Hmmm - It is a good point as well.

So here are my thoughts:
If list length == 1 or 0, all good. However if list.length > 1 then we need to know the reference for look up. 
What we can do is leave the return type to be an individual ucj, but adjust the logic here a little. 
Consider this: 
adding an input variable to getJunction - be reference id - then if list.length > 1, and reference(input variable) is empty then throw an exception for missing reference.
This can allow for some distinction in calling functions, and give us time to implement the functionality of these associations.

Reply:
How would this affect `getGrantPath`? While UCJs can be associated with references to other objects like this, getting a grant path with __0 or 1__ reference id seems like an arbitrary limitation; it may be desired to get the grant path in consideration of two different references for example.

Reply:
I have an idea where getGrantPath is passed something like:
```
[
  ["bankAccount", "asdf"],
  ["favouriteColour", "red"]
]
```

also indicating references are for specific capabilities may be useful

```
[
  ["capability-number-1", "favouriteColour", "red"],
  ["capability-number-2", "favouriteColour", "blue"]
]
```

Reply:
To clarify, I'm thinking of the two above as not mutually exclusive, so for example you could specify:
```
[
  ["bankAccount", "asdf"],
  ["capability-number-1", "favouriteColour", "red"],
  ["capability-number-2", "favouriteColour", "blue"]
]
```

Reply:
IMO we should merge this now and add this functionality as needed

Reply:
For @Nauna  solution we'd also have to pass in a DAOkey since we'd want to have the id reference any object on the system. We could have both a getJunction and getJunctions method